handlers/create.go

```diff
@@ -216,89 +219,42 @@ func getBaseMachineDir(uuid string) (string, error) {
-		}
-		if machine.Amazonec2Config.Ami != "" {
-			cmd = append(cmd, "--amazonec2-ami", machine.Amazonec2Config.Ami)
-		}
-		if machine.Amazonec2Config.SessionToken != "" {
-			cmd = append(cmd, "--amazonec2-session-token", machine.Amazonec2Config.SessionToken)
-		}
-		if machine.Amazonec2Config.Region != "" {
-			cmd = append(cmd, "--amazonec2-region", machine.Amazonec2Config.Region)
-		}
-		if machine.Amazonec2Config.Zone != "" {
-			cmd = append(cmd, "--amazonec2-zone", machine.Amazonec2Config.Zone)
-		}
-		if machine.Amazonec2Config.SubnetId != "" {
-			cmd = append(cmd, "--amazonec2-subnet-id", machine.Amazonec2Config.SubnetId)
-		}
-		if machine.Amazonec2Config.SecurityGroup != "" {
-			cmd = append(cmd, "--amazonec2-security-group", machine.Amazonec2Config.SecurityGroup)
-		}
-		if machine.Amazonec2Config.InstanceType != "" {
-			cmd = append(cmd, "--amazonec2-instance-type", machine.Amazonec2Config.InstanceType)
-		}
-		if machine.Amazonec2Config.RootSize != "" {
-			cmd = append(cmd, "--amazonec2-root-size", machine.Amazonec2Config.RootSize)
-		}
-		if machine.Amazonec2Config.IamInstanceProfile != "" {
-			cmd = append(cmd, "--amazonec2-iam-instance-profile", machine.Amazonec2Config.AccessKey)
-		}
-	case "virtualbox":
-		cmd = append(cmd, "virtualbox")
-		if machine.VirtualboxConfig.Boot2dockerUrl != "" {
-			cmd = append(cmd, "--virtualbox-boot2docker-url", machine.VirtualboxConfig.Boot2dockerUrl)
-		}
-		if machine.VirtualboxConfig.DiskSize != "" {
-			cmd = append(cmd, "--virtualbox-disk-size", machine.VirtualboxConfig.DiskSize)
-		}
-		if machine.VirtualboxConfig.Memory != "" {
-			cmd = append(cmd, "--virtualbox-memory", machine.VirtualboxConfig.Memory)
+	sDriver := strings.ToLower(machine.Driver)
+	cmd := []string{"create", "-d", sDriver}
+
+	rMachine := reflect.ValueOf(machine).Elem()
+
+	// Grab the reflected Value of XyzConfig (i.e. DigitaloceanConfig) based on the machine driver
+	rConfig := rMachine.FieldByName(strings.ToUpper(sDriver[:1]) + sDriver[1:] + "Config")
+	tConfig := rConfig.Type()
+
+	for i := 1; i < rConfig.NumField(); i++ {
+		// We are ignoring the Resource Field since it always appears first.  If it doesn't,
+		// we will need to check for name and ignore instead.
+		f := rConfig.Field(i)
+
+		// This converts all field name of ParameterName to --<driver name>-parameter-name
+		// i.e. AccessToken parameter for DigitalOcean driver becomes --digitalocean-access-token
+		dmField := "--" + sDriver + "-" + strings.ToLower(RegExHyphen.ReplaceAllString(tConfig.Field(i).Name, `${1}-${2}`))
+
+		// For now, we only support bool and string.  Will add more as required.
+		switch f.Interface().(type) {
+		case bool:
+			// dm only accepts field or field=true if value=true
+			if f.Bool() {
+				cmd = append(cmd, dmField)
+			}
+		case string:
+			if f.String() != "" {
+				cmd = append(cmd, dmField, f.String())
+			}
+		default:
+			log.Errorf("Unsupported type: %v", f.Type())
```

Comment:
I think we should do more than just log an error. I think we should return and error and let it bubble all the way up.